Hazelcast.Net/Hazelcast.Client.Proxy/ClientMapNearCacheProxy.cs

```diff
@@ -256,45 +258,46 @@ protected override object ExecuteOnKeyInternal(IData keyData, IEntryProcessor en
             }
         }
 
-        protected override void GetAllInternal(ICollection<IData> keyDatas, List<object> resultingKeyValuePairs)
+        protected override void GetAllInternal(ArrayList partitionToKeyData, ConcurrentQueue<KeyValuePair<IData, object>> resultingKeyValuePairs)
         {
-            var list = new List<IData>(keyDatas);
-            foreach (var keyData in keyDatas)
+            Parallel.For(0, partitionToKeyData.Count, partitionId =>
             {
-                object value;
-                if (_nearCache.TryGetValue(keyData, out value))
+                var keyList = (ArrayList)partitionToKeyData[partitionId];
+                for (int i = 0; i < keyList.Count; i++)
                 {
-                    list.Remove(keyData);
-                    resultingKeyValuePairs.Add(keyData);
-                    resultingKeyValuePairs.Add(value);
+                    var keyData = (IData)keyList[i];
+                    object value;
+                    if (_nearCache.TryGetValue(keyData, out value))
+                    {
+                        keyList.Remove(keyData);
```

Comment:
We are iterating over the keyList with indexes and calling remove in the middle. It looks like we will have array index out of bound exception.

Reply:
reverted the iteration